tests/integration/kubernetes/kata-deploy-test-usig-nfd.bats

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	node_name="$(kubectl get node -o name)"
+	pod_name="busybox"
+	nfd_tee_pod_name="nfd-tee-test"
+
+	get_pod_config_dir
+	get_tee
+	get_tee_key_resource
+	get_tee_key_label_jsonpath
+}
+
+@test "USING_NFD: true | Ensure the NodeFeatureRule is properly created" {
```

Comment:
IIUC you are going to create GHA jobs to USING_NFD true and false configurations. And for each TEE supported too? I'm asking because if we create jobs for each possible configuration we will end up with a lot of jobs (== welcome to hell)

In your opinion is it relevant to run the entire k8s suite for each case of USING_NFD true and false? If not, maybe we can change this test case to simply test if kata-deploy gets installed correctly. For example, for this test case (`USING_NFD: true | Ensure the NodeFeatureRule is properly created`):
 1. delete the kata-deploy
 2. Install kata-deploy with "USING_NFD=true" AND if GHA running on TDX. Otherwise skip.
 3.  Start a simple pod
 4. run `[[ $(kubectl get nodefeaturerule -o name | grep ${tee}-total-keys) ]]`